torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4342,6 +4377,24 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            op=torch.fmin,
            dtypes=all_types_and(torch.float16, torch.bfloat16, torch.bool),
            sample_inputs_func=sample_inputs_max_min_binary,),
+    OpInfo('fmod',
+           dtypes=all_types_and(torch.float16, torch.bool),
+           sample_inputs_func=sample_inputs_fmod_remainder),
+    OpInfo('fmod',
+           variant_test_name='autodiffed_args',
+           dtypes=all_types_and(torch.float16, torch.bool),
+           assert_autodiffed=True,
+           sample_inputs_func=partial(sample_inputs_fmod_remainder, autodiffed_args=True)),
+    OpInfo('remainder',
+           dtypesIfCPU=all_types_and(torch.float16, torch.bool),
+           dtypesIfCUDA=all_types_and(torch.float16, torch.bool, torch.bfloat16),
+           sample_inputs_func=sample_inputs_fmod_remainder),
+    OpInfo('remainder',
+           variant_test_name='autodiffed_args',
+           dtypesIfCPU=all_types_and(torch.float16, torch.bool),
+           dtypesIfCUDA=all_types_and(torch.float16, torch.bool, torch.bfloat16),
```

Comment:
This is interesting to me, I'm curious if the tensors in the numerator are promoted based on the dtype of denominator? (I'll also take a look at this tomorrow and see if I can find an answer)

```
>>> torch.remainder(torch.tensor((), dtype=torch.bool, device='cpu', requires_grad=False), torch.tensor(1, dtype=torch.bool))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: "remainder_cpu" not implemented for 'Bool'
>>> torch.remainder(torch.tensor((), dtype=torch.bool, device='cpu', requires_grad=False), 1.5)
tensor([])
```

Similarly for `fmod`:

```
>>> torch.fmod(torch.tensor((), dtype=torch.bool, device='cpu', requires_grad=False), 1.5)
tensor([])
>>> torch.fmod(torch.tensor((), dtype=torch.bool, device='cpu', requires_grad=False), torch.tensor(1, dtype=torch.bool))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: "fmod_cpu" not implemented for 'Bool'
```

Reason why `torch.bool` is included in `autodiffed_args` variant of `remainder/fmod`:

1. `fmod` and `remainder` support boolean tensors as numerators but _only_ integer and floating tensors (or scalars) as denominators.
2. Since inputs for `assert_autodiffed=True` only include scalars as denominators, hence `torch.bool` doesn't give a `RuntimeError`. That's the reason why `dtypesIfCPU` and `dtypesIfCUDA` contain `torch.bool`.

Reply:
Since we're talking about implementations, is this function used:

https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/aten/src/THC/generic/THCTensorMathPairwise.cu#L27

On CPU the function correctly supports type promotion based on the computation type:

https://github.com/pytorch/pytorch/blob/098d9975a7c6b33ada382bd51f322f7f5c694a17/aten/src/ATen/native/cpu/BinaryOpsKernel.cpp#L784

and on CUDA too:

https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/aten/src/ATen/native/cuda/BinaryRemainderKernel.cu#L39

If the THC function isn't used we should delete it as part of this PR

Reply:
Thanks for pointing it out. @mruberry - I've removed the THC function as a part of this PR.

Reply:
One interesting behavior though. When the `other` tensor is of floating type and `input` tensor is of integral type, `fmod` throws an error:

```
>>> torch.fmod(torch.tensor(6), torch.tensor(5.0))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: result type Float can't be cast to the desired output type Long
```

While, this works for `remainder`:

```
>>> torch.remainder(torch.tensor(6), torch.tensor(5.0))
tensor(1.)
```

Is this expected behavior?